OSU-Sustainability-Office/energy-dashboard
# Map Popup Adjustment for Smaller Laptop Screens

Issue Referenced: https://github.com/OSU-Sustainability-Office/energy-dashboard/issues/190

Testing procedure:
1. Start webpage locally.

2. While browser is in fullscreen mode, enter browser, Inspect Element > Console, then type `console.log(window.innerHeight)`

    - For me I have 722 for `window.innerHeight` value. If your value is different, you may have to adjust the `Vue.prototype.$WINDOW_HEIGHT` value in `src/main.js` to be higher or lower before proceeding. 
    - For testing purposes, the `$WINDOW_HEIGHT` value should be slightly lower than what your actual `window.innerHeight` value is (I set `$WINDOW_HEIGHT` to 700 which is a bit below 722).

3. Return to browser in fullscreen mode, click on a building and observe popup.

4. Close the popup. 

5. Then set browser to windowed mode with a `window.innerHeight` value lower than your `$WINDOW_HEIGHT` value (check `window.innerHeight` in dev console to be sure), click on a building and observe popup.

Notes:
- After you adjust the browser window size, you need to close the current popup and re-open it for the "map popup mode" to adjust.
- Reloading the webpage is not necessary
- Hard-coded values I'm not sure about I've commented about in the code.
- `$WINDOW_HEIGHT` will need to be adjusted (see edit below).

Screenshots:
full screen (`window.innerHeight` = 722)
![full screen](https://media.discordapp.net/attachments/1018323831468851202/1055296861440847882/image.png?width=1298&height=670)

windowed (`window.innerHeight` = 612)
![windowed](https://media.discordapp.net/attachments/1018323831468851202/1055297037719048192/image.png?width=1440&height=650)

EDIT 12/24/2022:
After testing with a page ruler chrome extension (https://chrome.google.com/webstore/detail/page-ruler/jcbmcnpepaddcedmjdcmhbekjhbfnlff), I have concluded that we need a `$WINDOW_HEIGHT` value of at least 866, 900 to be safe, in order for the "tall mode" of the map popup to fully show up without need of scrolling. Window inner height measures from top of website navbar to the windows navbar below (or whatever the equivalent on other OS).

The `$WINDOW_HEIGHT` value of 700 currently is just for testing, will need to be adjusted.

- Note: Testing with ruler above was done by clicking on the map popup, then zooming out. If you don't click on the popup first, the popup will not "zoom out with you". 
- Zooming out makes the ruler detect more pixel length (and seems accurate), but it doesn't change the value you get when typing `console.log(window.innerHeight)` into browser dev console. Due to this I (or someone else) will have to test this sometime on a bigger monitor (mine only supports up to 1920 x 1080 screen resolution).

Reference Image:
![Reference Image](https://media.discordapp.net/attachments/1018323831468851202/1056376092040110151/image.png?width=1421&height=670)
